### PR TITLE
ProtoData-based validation

### PR DESCRIPTION
**Pull request by dmdashenkov**
In this PR we integrate the new ProtoData-based validation with the model compiler.

Previously, validation code used to be generated via a Protoc plugin.
Now, we have a separate [library](https://github.com/SpineEventEngine/validation/) for that, built on top of ProtoData.

The old validation code generator, as well as the runtime message validator, are still present in this repository. We will later remove this code, alongside moving validation tests to the `validation` repo.

Also, in this PR we disable incremental build features for rejection generation. This config has caused us a lot of pain because of a weird bug when the task is considered up-to-date after the project cleaning. It's not clear if this is our fault or Gradle's. Previously, we manually removed the `.gradle` dir to make the build work again. Hopefully, now we won't have to.

### CODE REVIEW

File: tests/buildSrc/src/main/kotlin/io/spine/internal/gradle/javascript/JsContext.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * Executes `npm` command in a separate process.
+     *
+     * This [File] is used as a working directory.
+     */
+    fun File.npm(vararg args: String) = project.exec {
+
+        workingDir(this@npm)
+        commandLine(npmExecutable)
+        args(*args)
```
[the remark this answers is not in the document]
Comment (dmdashenkov) [reply tests/buildSrc/src/main/kotlin/io/spine/internal/gradle/javascript/JsContext.kt:54]: ignore

File: tests/buildSrc/src/main/kotlin/io/spine/internal/gradle/dart/DartContext.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+open class DartContext(dartEnv: DartEnvironment, internal val project: Project)
+    : DartEnvironment by dartEnv
+{
+    /**
+     * Executes `pub` command in this [Exec] task.
+     *
+     * The Dart ecosystem uses packages to manage shared software such as libraries and tools.
+     * To get or publish Dart packages, the `pub` package manager is to be used.
+     */
+    fun Exec.pub(vararg args: Any) = commandLine(pubExecutable, *args)
```
[the remark this answers is not in the document]
Comment (dmdashenkov) [reply tests/buildSrc/src/main/kotlin/io/spine/internal/gradle/dart/DartContext.kt:44]: ignore

File: mc-java/src/main/java/io/spine/tools/mc/java/gradle/plugins/GenerateProtoDataConfig.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import static java.nio.file.Files.write;
+import static java.util.stream.Collectors.toSet;
+
+public abstract class GenerateProtoDataConfig extends DefaultTask {
+
+    @OutputFile
+    public abstract RegularFileProperty getTargetFile();
+
+    @TaskAction
+    private void writeFile() throws IOException {
```
[the remark this answers is not in the document]
Comment (dmdashenkov) [reply mc-java/src/main/java/io/spine/tools/mc/java/gradle/plugins/GenerateProtoDataConfig.java:67]: ignore

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov, @armiol, PTAL.

File: tests/validation-gen/src/test/java/io/spine/test/tools/validate/RequiredFieldConstraintTest.java [hunk 4]
```diff
@@ -26,66 +26,69 @@
...
+        var invalidMessage = Due.newBuilder();
         assertInvalid(invalidMessage, "date | never");
     }
 
     @Test
     @DisplayName("a complete group of fields must be set")
     void notComplete() {
-        var invalidMessage = Combination.newBuilder()
+        var invalidMessage = Combination
+                .newBuilder()
```
Comment (armiol) [inline tests/validation-gen/src/test/java/io/spine/test/tools/validate/RequiredFieldConstraintTest.java:56]: Please move this `.newBuilder()` in this case — and in other cases in this file — a line back.

File: tests/validation-gen/src/test/java/io/spine/test/tools/validate/ValidateConstraintTest.java [hunk 5]
```diff
@@ -42,13 +40,19 @@
     @Test
     @DisplayName("message fields are validated")
     void checkEnclosedFields() {
-        var wrongAddress = DeliveryReceiver.newBuilder()
+        var wrongAddress = DeliveryReceiver
+                .newBuilder()
```
Comment (armiol) [inline tests/validation-gen/src/test/java/io/spine/test/tools/validate/ValidateConstraintTest.java:44]: Please move `.newBuilder()` a line back here and in the similar cases in this file.

Also, the expressions down below should be re-formatted once more.

**Review by armiol — APPROVED** [review]
@dmdashenkov LGTM with some comments to address.

**Pull request merged**